provider/disk_utils.py

```diff
@@ -0,0 +1,159 @@
+"""
+Disk utilities function for managing disks and filesystems in guest environments.
+
+This module contains a function related to disk I/O test command execution on
+different OS types.
+"""
+
+from virttest import utils_disk
+from virttest.utils_misc import get_linux_drive_path
+from virttest.utils_windows.drive import get_disk_props_by_serial_number
+
+
+def get_disk_by_serial(os_type, serial, session):
+    """
+    This function provides the disk ID reference.
+    It will vary based on the OS (ID or path).
+
+    :param os_type: The VM Operating System.
+    :param disk_id: The VM disk ID defined in the test cfg.
+    :param session: The guest session.
+    """
+    if os_type == "windows":
+        idx_info = get_disk_props_by_serial_number(session, serial, ["Index"])
+        if idx_info:
+            return idx_info["Index"]
+    else:
+        return get_linux_drive_path(session, serial)
+
+
+def init_disk_by_id(
+    params,
+    vm,
+    fstype=None,
+    os_type=None,
+    did="",
+):
+    """
+    Initializes the disk in the guest based on the received device ID.
+    By initialize, it means obtain the reference of the disk (ID or path) and
+    later in Linux systems, create the filesystem + mounting or formatting the
+    disk in Windows ones.
+
+    :param params: Dictionary with the test parameters.
+    :param vm: The VM object.
+    :param fstype: The kind of filesystem (ntfs or xfs).
+    :param os_type: The VM Operating System.
+    :param did: The ID of the VM's disk.
+    """
+    session = vm.wait_for_login()
+    dst_dir = params.get("dst_dir")
+    image_params = params.object_params(did)
+    img_size = image_params.get("image_size")
+
+    disk = get_disk_by_serial(os_type, did, session)
+
+    if os_type == "linux":
+        partition_name = disk.split("/")[2]
+        utils_disk.create_filesyetem_linux(session, partition_name, fstype="xfs")
+        utils_disk.mount(disk, dst_dir, fstype, session=session)
```

Comment:
We hard code to create the xfs type files system for the linux guest, but we could mount the disk with another fstype, I am not sure if it meets the test requirements @qingwangrh .